packages/samples/headless-commerce-ssr-remix/client/external-catalog-api.ts

```diff
@@ -0,0 +1,36 @@
+export type ExternalCatalogItem = {
```

Comment:
The implementation of this "API" is completely silly, but the idea is to show clearly that the main product information shown on a PDP should typically come from some external service, not from a Coveo query.

Reply:
This is not in the current next sample and I don't think it is necessary imo. This is different from the cart since it has 0 relation to coveo except for it to potentially use the same productId. I think we should remove this.

Reply:
We can keep it but can we add some comments ? Similar to here ?
https://github.com/coveo/ui-kit/blob/master/packages/samples/headless-ssr-commerce/actions/external-cart-api.ts